intro_to_R_tidyverse/intro_to_r.Rmd

```diff
@@ -0,0 +1,464 @@
+an object so, for a vector that has 10 pieces dimensions will be shown as 
+`[1:10]`.
+```{r Vectors again}
+one_through_ten <- 1:10
+str(one_through_ten)
+```
+
+A matrix is a two dimensional collection of data. Like in math, commas separate
+rows and columns indices. 
+We will generally be using matrices when we are working with only the numbers of
+a gene expression matrix.
+eg. for a vector that has 10 rows and 10 columns `str()` will report dimensions 
+of `[1:10, 1:10]`
+```{r Matrix}
+# Note that we need to use arguments for the matrix function to tell R how many 
+# rows "nrow = 10" and how many columns "ncol = 10" we want. 
+sq_matrix <- matrix(1:100, nrow = 10, ncol = 10)
+str(sq_matrix)
+```
+
+What dimensions do you suppose this matrix will have by default if we don't 
+specify the number of rows and columns we want? 
+```{r Matrix cont.}
+not_sq_matrix <- matrix(1:100)
+str(not_sq_matrix)
+```
+
+Matrices however, are not great at storing different types of data. 
+Notice that in the above examples we are only using integers. 
+If you try to mix data types together in a matrix, R will coerce them to be all 
+one type, which will probably be bad for you because your information will be 
+distorted.   
+  
+Hence the `data.frame` is probably what you will be using for a lot of your data. 
+Working with `data.frame`s also is handy because Hadley Wickham's `tidyverse` tools 
+are mostly built around using `data.frame`s or its' closely related sibling the 
+`tibble`. The next Rmd will show you all about manipulating `data.frame`s with the 
+tidyverse tools. 
+```{r Data frames}
+# Let's make a data frame from our previously created vectors!
+df <- data.frame(nums , words, color_vector)
+```
+
+Note that if these vectors were not conveniently all the same length, we could 
+not put them in a `data.frame` together.  
+```{r}
+str(df)
+# Also note that "obs" are rows and "variables" are the columns in a data.frame
+```
+
+Also note that the `data.frame` function makes strings a factor by default,
+unless you tell it not to. 
+So let's remake this `data.frame`, but tell R not to do that with one of the 
+arguments for the `data.frame` function. 
+```{r Data frames cont. }
+# Let's make a data frame from our previously created vectors but tell R not to
+# turn strings into factors!
+df <- data.frame(nums, words, color_vector, stringsAsFactors = FALSE)
+str(df)
+# Note that with the argument `stringsAsFactors` now our words variable remains 
+# a character vector as it was before. 
+```
+
+Now that we have a nice `data.frame` and other objects to play with in our 
+environment; let's play around with it and extract pieces of it. 
+```{r Subsetting Indices}
+# Let's say we only want to see what's in the first row and first column of this
+# data.frame. We can subset data structures by using brackets. 
+df[1, 1]
+
+# Note that you can do this same thing to matrices: 
+sq_matrix[3, 3]
+
```

Comment:
Can we add one that uses `:` -- such as `sq_matrix[, 1:3]`? This has an added benefit in that it demonstrates that all rows will be selected.

Reply:
Added it.